Src/Newtonsoft.Json/Converters/IsoDateTimeConverter.cs

```diff
@@ -57,7 +57,7 @@ public DateTimeStyles DateTimeStyles
         public string DateTimeFormat
         {
             get { return _dateTimeFormat ?? string.Empty; }
-            set { _dateTimeFormat = StringUtils.NullEmptyString(value); }
+            set { _dateTimeFormat = (string.IsNullOrEmpty(value)) ? null : value; }
```

Comment:
Curious: How does this change fit? And why the extra parentheses around the condition?